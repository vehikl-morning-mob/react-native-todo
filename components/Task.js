import React from 'react';
import {
  StyleSheet,
  Text,
  TouchableHighlight
} from 'react-native';


const Task = ({ task, toggleCompleted}) => {
    return (
      <TouchableHighlight onPress={() => toggleCompleted(task)}>
        <Text style={task.completed && styles.textStrikeThrough}>{task.name}</Text>
      </TouchableHighlight>
    )
}

const styles = StyleSheet.create({
  textStrikeThrough: {
    textDecorationLine: 'line-through'
  },
});

export default Task;
