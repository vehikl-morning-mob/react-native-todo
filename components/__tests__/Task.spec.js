import React from 'react';
import renderer from 'react-test-renderer';
import Task from '../Task';
import { TouchableHighlight } from 'react-native';
import { isTemplateElement } from '@babel/types';

describe('Task', () => {
  it('displays the task', () => {
    const task = {
      id: 1,
      name: 'Foo',
      completed: false,
    }
    const component = renderer.create(<Task task={ task } />).toJSON();
    expect(component).toMatchSnapshot();
  })

  it('displays completed task', () => {
    const task = {
      id: 1,
      name: 'Foo',
      completed: true,
    }
    const component = renderer.create(<Task task={task} />).toJSON();
    expect(component).toMatchSnapshot();
  })

  it('toggles completed task on click', () => {
    const task = {
      id: 1,
      name: 'Foo',
      completed: false,
    }
    const toggleCompleted = jest.fn(() => {
      task.completed = true;
    });
    const component = renderer.create(<Task task={task} toggleCompleted={toggleCompleted} />).root;

    component.findByType(TouchableHighlight).props.onPress();

    expect(toggleCompleted).toHaveBeenCalledWith(task);
    expect(task.completed).toBe(true);
  })
})