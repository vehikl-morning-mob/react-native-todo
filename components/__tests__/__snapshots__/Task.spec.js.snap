// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task displays completed task 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <Text
    style={
      Object {
        "textDecorationLine": "line-through",
      }
    }
  >
    Foo
  </Text>
</View>
`;

exports[`Task displays the task 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <Text
    style={false}
  >
    Foo
  </Text>
</View>
`;
